app/services/user_signup.rb

```diff
@@ -0,0 +1,50 @@
+class UserSignup
+  
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def signup(options={})
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(
+        :amount => 999,
+        :source => options[:stripeToken]
+        )
+      if charge.successful?
+        @user.save
+        AppMailer.welcome_email(@user).deliver
+        if options[:invite_token]
+          handle_invited_user(options[:invite_token])
+          @status = :successful
+          self
+        else
+          @status = :successful
+          self
+        end
+      else
+        @status = :failed
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :failed
+      @error_message = "Please fill out all user fields"
```

Comment:
This service might be used with sources other than the form. In a production app you might want a more general message that just says there was a problem and please try again. That would apply to every data source, rather than just a form. Just something to think about :smile: